etl/src/v2/conversions/event.rs

```diff
@@ -0,0 +1,374 @@
+            .ok_or(EventConversionError::MissingSchema(table_id))
+    }
+
+    fn convert_tuple_to_row(
+        column_schemas: &[ColumnSchema],
+        tuple_data: &[TupleData],
+    ) -> Result<TableRow, EventConversionError> {
+        let mut values = Vec::with_capacity(column_schemas.len());
+
+        for (i, column_schema) in column_schemas.iter().enumerate() {
+            let cell = match &tuple_data[i] {
+                TupleData::Null => Cell::Null,
+                TupleData::UnchangedToast => TextFormatConverter::default_value(&column_schema.typ),
+                TupleData::Binary(_) => return Err(EventConversionError::BinaryFormatNotSupported),
+                TupleData::Text(bytes) => {
+                    let str = str::from_utf8(&bytes[..])?;
+                    TextFormatConverter::try_from_str(&column_schema.typ, str)?
+                }
+            };
+            values.push(cell);
+        }
+
+        Ok(TableRow { values })
+    }
+
+    async fn convert_insert_to_event(
+        &self,
+        insert_body: &InsertBody,
+    ) -> Result<Event, EventConversionError> {
+        let table_id = insert_body.rel_id();
+        let table_schema = self.get_table_schema(table_id).await?;
+        let row = Self::convert_tuple_to_row(
+            &table_schema.column_schemas,
+            insert_body.tuple().tuple_data(),
+        )?;
+
+        Ok(Event::Insert(InsertEvent { table_id, row }))
+    }
+
+    async fn convert_update_to_event(
+        &self,
+        update_body: &UpdateBody,
+    ) -> Result<Event, EventConversionError> {
+        let table_id = update_body.rel_id();
+        let table_schema = self.get_table_schema(table_id).await?;
+        let identity = update_body
+            .key_tuple()
+            .or(update_body.old_tuple())
+            .ok_or(EventConversionError::MissingTupleInDeleteBody)?;
+
+        let identity_row =
+            Self::convert_tuple_to_row(&table_schema.column_schemas, identity.tuple_data())?;
+        let row = Self::convert_tuple_to_row(
+            &table_schema.column_schemas,
+            update_body.new_tuple().tuple_data(),
+        )?;
+
+        Ok(Event::Update(UpdateEvent {
+            table_id,
+            row,
+            identity_row,
+        }))
+    }
+
+    async fn convert_delete_to_event(
+        &self,
+        delete_body: &DeleteBody,
+    ) -> Result<Event, EventConversionError> {
+        let table_id = delete_body.rel_id();
+        let table_schema = self.get_table_schema(table_id).await?;
+        let identity = delete_body
+            .key_tuple()
+            .or(delete_body.old_tuple())
+            .ok_or(EventConversionError::MissingTupleInDeleteBody)?;
+
+        let identity_row =
+            Self::convert_tuple_to_row(&table_schema.column_schemas, identity.tuple_data())?;
+
+        Ok(Event::Delete(DeleteEvent {
+            table_id,
+            identity_row,
+        }))
+    }
+
+    pub async fn convert(
```

Comment:
I copied similar logic from the previous event conversion but with the following differences:
- The state store is now used to get table schemas, which allows a single converter to pull it's own schema based on the incoming message.
- The new event types are using all of our own internal types, so that we have full control of their behavior.